### PR TITLE
[MRG+2] Fix mathtext parsing of combined super/sub-scripts issue #7615

### PR DESCRIPTION
**Pull request by rtrhd**
Fix for #7615.

Fully handle both subscript and superscript together. The issue related to the unhandled case of four tokens in subsuper() close to where the error was raised. This occurs when the _a^b follows a construct
like \left(X\right). There are some overwritten builtins in this function but I've left them as they are. 

I've added a test case and corresponding reference images.

### CODE REVIEW

**Comment by NelleV** [comment]
Hi @rtrhd 
The patch looks good, but this IMO should not target 2.x (we are in release candidate mode and this is not a critical bug).
Wait until confirmation by @tacaswell, but I would recommend targetting master.

Else, it looks good.

File: lib/matplotlib/mathtext.py [hunk 1]
```diff
@@ -2881,21 +2881,18 @@ def subsuper(self, s, loc, toks):
...
-        elif len(toks) == 2:
-            op, next = toks
-            nucleus = Hbox(0.0)
-            if op == '_':
-                sub = next
-            else:
-                super = next
-        elif len(toks) == 3:
-            nucleus, op, next = toks
+        elif len(toks) in (2, 3):
```
Comment (tacaswell) [inline lib/matplotlib/mathtext.py]: to check my understanding, this change is just condensing two cases into 1

File: lib/matplotlib/mathtext.py [hunk 2]
```diff
@@ -2881,21 +2881,18 @@ def subsuper(self, s, loc, toks):
...
+            # single subscript or superscript
+            nucleus = toks[0] if len(toks) == 3 else Hbox(0.0)
+            op, next = toks[-2:]
             if op == '_':
                 sub = next
             else:
                 super = next
-        elif len(toks) == 5:
-            nucleus, op1, next1, op2, next2 = toks
+        elif len(toks) in (4, 5):
```
Comment (tacaswell) [inline lib/matplotlib/mathtext.py]: and this change in the actually fixes the bug?

**Comment by tacaswell** [comment]
It is fine to target 2.x, we will just hold of merging until 2.0 is out.

**Comment by rtrhd** [comment]
Thanks both for the review. Yes, the first change was just to condense the two cases to match what I'd done with the bugfix.

**Pull request merged**